react-native-community/cli
# feat: add Bun to `init` command

Summary:
---------
Adding support for Bun package manager with `init` command.
With template and dependencies installed via bun package manager, the JS code execution time went from ~41s to ~2,5s.

In order to make package managers easier to use, this PR deprecates `--npm` flag and introduces `--pm` flag where it's possible to use one of the packagers: `yarn`, `npm` or `bun`.




Test Plan:
----------
Follow the contributing guide
1. Run `init` command as usual. It should install template and dependencies using yarn and you should have `yarn.lock` file in created project.
2. Run `init` command with `--pm yarn` flag.  Once it's done, you should have `yarn.lock` file in created project.
3. Run `init` command with `--pm npm` flag. Once it's done, you should have `package-lock.json` file in created project.
4. Run `init` command with `--pm bun` flag. Once it's done, you should have `bun.lockb` file in created project. 
5. Run `init` command with `--npm` flag. It should show a deprecation warning in the console.
6. Run `bun run ./path-to-cli init ProjectName`. It should use `bun` to install dependencies and template without any additional flag passed. Should also work with `bunx` once deployed.
7. If `bun` is not installed on the machine, it should ~~fallback to `npm`. Created project should have a `package-lock.json` file.~~ throw an error that selected package manager is not available (only when using flag)
8. Run `clean command` in the created project - the `Bun` option should be available. It runs `bun pm cache rm` under the hood.


Checklist
----------
